test_plus/test.py

```diff
@@ -381,6 +383,24 @@ def __init__(self, *args, **kwargs):
         self.client_class = get_api_client()
         super(APITestCase, self).__init__(*args, **kwargs)
 
+    # APITest requests now default to JSON
+    def request(self, method, url_name, *args, **kwargs):
+        data = kwargs.get("data")
+        kwargs["data"] = json.dumps(data) if data is not None else None
+        return getattr(self, method)(url_name, *args, **kwargs)
+
+    def post(self, url_name, *args, **kwargs):
+        return self.request("post", url_name, *args, **kwargs)
+
+    def get(self, url_name, *args, **kwargs):
+        return self.request("get", url_name, *args, **kwargs)
+
+    def put(self, url_name, *args, **kwargs):
+        return self.request("put", url_name, *args, **kwargs)
+
+    def delete(self, url_name, *args, **kwargs):
+        return self.request("delete", url_name, *args, **kwargs)
```

Comment:
These appear to be recursive btw. I think #390 might need to getattr(TestCase) instead of self but this might have been what you and @frankwiles were discussing during banter. 

You might also need `def head` and the other HTTP verbs we are missing here if this is the fix.